src/searchor/__init__.py

```diff
@@ -57,17 +58,8 @@ class Engine:
     Yahoo = "https://search.yahoo.com/search?p={query}"
     Yandex = "https://yandex.com/search/?text={query}"
 
+    def search(self, query, additional_queries:dict=None):
```

Comment:
Don't we need the `engine` parameter here as default set to`Engine.Google`? This would override the whole docs if so

Reply:
Not necessarily, it is handled by the Enum subclass. The examples would need to be modified accordingly, but not by a significant amount.
Ex:
`engine = Engine.Google`
`print(engine.search("hello")) #https://www.google.com/search?q=hello`
`print(Engine.Google.search("foo")) #https://www.google.com/search?q=foo`

Reply:
Alright! Sounds good.

Reply:
By handling the Engine as a proper Enum instead of just strings within a class, it allows for better structure and more options with scalability in the future. It also allows grouping the search function and the future functions made into the Engine class as a whole.

If this route would rather not be taken, we can still make the Engine an Enum, and handle the search function in another class.